Block/Adminhtml/System/Config/CardFormCustomization/js/script.js

```diff
@@ -0,0 +1,206 @@
+class CardFormCustomization {
+    init() {
+        this.handleButtonClickEvent()
+        this.handleColorInputChanges()
+        this.showOmiseCardForm()
+    }
+
+    showModal() {
+        const element = document.getElementById('omise-modal')
+        element.style.display = 'flex'
+        this.setDesignFormValues()
+    }
+
+    showPreviewModal() {
+        const element = document.getElementById('omise-preview-modal')
+        element.style.display = 'flex'
+    }
+
+    hidePreviewModal() {
+        const element = document.getElementById('omise-preview-modal')
+        element.style.display = 'none'
+    }
+
+    hideModal() {
+        const element = document.getElementById('omise-modal')
+        element.style.display = 'none'
+    }
+
+    getSelectedTheme() {
+        const cardThemeInputName = `groups\\[omise\\]\\[groups\\]\\[omise_cc\\]\\[fields\\]\\[card_form_theme\\]\\[value\\]`
+        return this.getInputValue(cardThemeInputName)
+    }
+
+    getInitialDesign() {
+        const design = this.parseJson(OMISE_CARD_CUSTOMIZATION_DESIGN)
+        const darkTheme = this.parseJson(OMISE_CARD_CUSTOMIZATION_DARK_THEME)
+        const lightTheme = this.parseJson(OMISE_CARD_CUSTOMIZATION_LIGHT_THEME)
+        if (!design) {
+            return this.getSelectedTheme() == 'dark' ? darkTheme : lightTheme
+        }
+        return design
+    }
+
+    setInputValue(name, val) {
+        document.querySelector(`[name=${name}]`).value = val
+    }
+
+    getInputValue(name) {
+        const element = document.querySelector(`[name=${name}]`);
+        if(element) {
+            return element.value
+        }
+        return null;
+    }
+
+    setColorInputValue(element) {
+        const valueElement = element.nextElementSibling
+        if (!valueElement) {
+            element.insertAdjacentHTML('afterend', `<span>${element.value}</span>`)
+        } else {
+            valueElement.innerHTML = element.value
+        }
+    }
+
+    handleColorInputChanges() {
+        const colorInputs = document.querySelectorAll('.omise-color-input-container')
+
+        colorInputs.forEach((element) => {
+            const input = element.querySelector('.omise-color-input')
+
+            this.setColorInputValue(input)
+            input.addEventListener('change', (event) => {
+                this.setColorInputValue(event.target)
+            })
+
+            element.addEventListener('click', (event) => {
+                let input = event.target.querySelector('.omise-color-input')
+                if (!input) {
+                    input = event.target.previousElementSibling
+                }
+                const clickEvent = new MouseEvent('click')
+                if(input) {
+                    input.dispatchEvent(clickEvent)
+                }
+            })
+        })
+    }
+
+    setDesignFormValues() {
+        const design = this.getInitialDesign()
+        const self = this
+        if (design) {
```

Comment:
What's the else condition for this?

Reply:
No else condition, it is just to prevent `Object.keys()` on null  throw error if `design` is null.